src/GDGUkraine/lib/tools/authorize.py

```diff
@@ -0,0 +1,38 @@
+import cherrypy
+
+
+__all__ = ['AuthorizeTool']
+
+
+class AuthorizeTool(cherrypy.Tool):
+
+    def __init__(self):
+        cherrypy.Tool.__init__(
+            self, 'before_handler', self._fetch, priority=20
```

Comment:
Maybe, `on_start_resource` is better? Plz check whether session is available at that hook point.